api/src/main/java/javax/enterprise/inject/spi/builder/Builders.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.spi.builder;
+
+import java.util.ServiceLoader;
+
+/**
+ * Provide static access to various CDI builders.
+ * Will load {@link BuildersProvider} implementation with {@link ServiceLoader}.
+ *
+ *
+ * @see BuildersProvider
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public final class Builders {
+
+    /**
+     * {@link BuildersProvider} instance loader from {@link ServiceLoader}
+     */
+    protected static volatile BuildersProvider buildersProvider = null;
+
+    /**
+     * Avoid instantiation of this service class
+     */
+    private Builders() {
+    }
+
+    /**
+     * Obtains an {@link AnnotatedTypeBuilder} for the given type
+     *
+     * @param type the type that the AnnotatedType to build will represent
+     * @param <T> generic for type
+     * @return a reusable AnnotatedTypeBuilder for the given type
+     */
+    public static <T> AnnotatedTypeBuilder<T> annotatedType(Class<T> type) {
+        return provider().getAnnotatedTypeBuilder(type);
+    }
+
+    /**
+     * Obtains a generic {@link AnnotatedTypeBuilder}
+     *
+     * @return a reusable AnnotatedTypeBuilder
+     */
+    public static AnnotatedTypeBuilder annotatedType() {
+        return provider().getAnnotatedTypeBuilder();
+    }
+
+    /**
+     * Obtains a {@link InjectionPointBuilder}
+     *
+     * @return a reusable InjectionPointBuilder
+     */
+    public static InjectionPointBuilder injectionPoint() {
+        return provider().getInjectionPointBuilder();
+    }
+
+    /**
+     * Obtains a {@link BeanAttributesBuilder} for the given type
+     *
+     * @param type the type of the instance of the configured Bean
+     * @param <T> generic for type
+     * @return a reusable BeanAttributesBuilder
+     */
+    public static <T> BeanAttributesBuilder<T> beanAttributes(Class<T> type) {
+        return provider().getBeanAttributesBuilder(type);
+    }
+
+    /**
+     * Obtains a generic {@link BeanAttributesBuilder}
+     *
+     * @return a reusable BeanAttributesBuilder
+     */
+    public static  BeanAttributesBuilder beanAttributes() {
+        return provider().getBeanAttributesBuilder();
+    }
+
+    /**
+     * Obtains a {@link BeanBuilder} for the given type
+     *
+     * @param type the type of the instance of the built Bean
+     * @param <T> generic for type
+     * @return a reusable BeanAttributesBuilder
+     */
+    public static <T> BeanBuilder<T> bean(Class<T> type) {
+        return provider().getBeanBuilder(type);
+    }
+
+    /**
+     * Obtains a generic {@link BeanBuilder}
+     *
+     * @return a reusable BeanBuilder
+     */
+    public static BeanBuilder bean() {
```

Comment:
You'll need these raw methods when working with unknown type.

How can know the parameter type for the AnnotatedTypeBuilder if you are in:

`public void example(@Observes ProcessAnnotatedType<?> pat)`

method?

The need is the same for `BeanAttributesBuilder` and `ObserverMethodBuilder` since they can be used in observers with wildcard in their parameter type and a `set` method

It is perhaps useless with `BeanBuilder` since it will be used only in AfterBeanDiscovery and no type parameter enforcement.